third_party/openocd_cmake/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+cmake_minimum_required(VERSION 3.15)
+
+if (${CMAKE_VERSION} VERSION_GREATER "3.8")
+    #For cmake >= 3.9 INTERPROCEDURAL_OPTIMIZATION behaviour we need to explicitly
+    #set the cmake policy version number
+    cmake_policy(VERSION 3.9) 
+
+    # If we are using verison < 3.9 then setting INTERPROCEDURAL_OPTIMIZATION
+    # has no effect unless an Intel compiler is used
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+    message(WARNING "Build type not set, falling back to Release mode.
+        To specify build type use:
+        -DCMAKE_BUILD_TYPE=<mode> where <mode> is Debug or Release.")
+    set(CMAKE_BUILD_TYPE
+        "Release"
+        CACHE STRING "Choose the type of build, options are: Debug Release."
+        FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+project(rs_openocd)
+
+if(RAPTOR)
+    set(root_dir ${PROJECT_SOURCE_DIR}/../../../..)
+else()
+    if ((EXISTS ${PROJECT_SOURCE_DIR}/../../build) OR (EXISTS ${PROJECT_SOURCE_DIR}/../../dbuild))
+        set(root_dir ${PROJECT_SOURCE_DIR}/../..)
+    else()
+        set(root_dir ${PROJECT_SOURCE_DIR}/../../..)
+    endif()
+endif()
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(build_dir ${root_dir}/dbuild)
+else ()
+    set(build_dir ${root_dir}/build)
+endif()
+
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+    # <TODO>
+    #set(rs_openocd_tar "rs_openocd-win.zip")
+else()
+    set(rs_openocd_tar "openocd-7f5a821-linux86_64.tar.gz")
+endif()
+
+if(RAPTOR)
+    set(Destination_Dir ${build_dir}/share/raptor/configuration)
+else()
+    if ((EXISTS ${PROJECT_SOURCE_DIR}/../../build) OR (EXISTS ${PROJECT_SOURCE_DIR}/../../dbuild))
+        set(Destination_Dir ${build_dir}/share/foedag/configuration)
+    else()
+        set(Destination_Dir ${build_dir}/share/raptor/configuration)
+        set(root_dir ${PROJECT_SOURCE_DIR}/../../..)
+    endif()
+endif()
+
+set(download_path "https://github.com/RapidSilicon/openocd/releases/download/latest/${rs_openocd_tar}")
+if(WIN32)
+    # <TODO> - set temp_path to rs_openocd zip file when windows build available
+    # set(temp_path "${CMAKE_CURRENT_SOURCE_DIR}/rs_openocd-win.zip")
+else()
+    set(temp_path ${CMAKE_CURRENT_SOURCE_DIR}/openocd-7f5a821-linux86_64.tar.gz)
+endif()
+
+set(bin_path ${build_dir}/bin)
+
+if(NOT EXISTS ${temp_path})
+    file(DOWNLOAD ${download_path} ${temp_path})
+endif()
+
+add_library(rs_openocd)
```

Comment:
Are we using it somewhere?

Reply:
ops. I think line72 should be deleted. I will clean up the code.